shadows/framework/src/main/java/org/robolectric/shadows/ShadowBiometricManager.java

```diff
@@ -37,23 +37,67 @@ public class ShadowBiometricManager {
   protected int canAuthenticate() {
     if (RuntimeEnvironment.getApiLevel() >= R) {
       return reflector(BiometricManagerReflector.class, realBiometricManager).canAuthenticate();
+    } else if (canAuthenticateInternal(0, BiometricManager.Authenticators.BIOMETRIC_WEAK)
+        != BiometricManager.BIOMETRIC_SUCCESS) {
+      return canAuthenticateInternal(0, BiometricManager.Authenticators.BIOMETRIC_WEAK);
     } else if (biometricServiceConnected) {
-      return BIOMETRIC_SUCCESS;
+      return BiometricManager.BIOMETRIC_SUCCESS;
     } else {
-      boolean hasBiomatrics =
+      boolean hasBiometrics =
           ReflectionHelpers.callStaticMethod(
               BiometricManager.class,
               "hasBiometrics",
-              ClassParameter.from(
+              ReflectionHelpers.ClassParameter.from(
                   Context.class, RuntimeEnvironment.getApplication().getApplicationContext()));
-      if (!hasBiomatrics) {
-        return BIOMETRIC_ERROR_NO_HARDWARE;
+      if (!hasBiometrics) {
+        return BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE;
       } else {
-        return BIOMETRIC_ERROR_HW_UNAVAILABLE;
+        return BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE;
       }
     }
   }
 
+  @RequiresPermission(USE_BIOMETRIC)
+  @Implementation
+  protected int canAuthenticate(int authenticators) {
+    return canAuthenticateInternal(0, authenticators);
+  }
+
+  @RequiresPermission(USE_BIOMETRIC)
```

Comment:
What about adding a test to check permission requirement?

Reply:
Sure , i can add it, do you know any example i can reference on this part ?

Reply:
Looks like there don't exist tests to test those scenarios. I will resolve it.